domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasNoFractionalPartRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,218 @@
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputHasNoFractionalPartRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputHasNoFractionalPartRuleClassifierProviderTest {
+
+  private val WHOLE_NUMBER_123 = createWholeNumber(isNegative = false, value = 123)
+  private val FRACTION_2_OVER_4 = createFraction(isNegative = false, numerator = 2, denominator = 4)
+  private val FRACTION_1_OVER_2 = createFraction(isNegative = false, numerator = 1, denominator = 2)
+  private val MIXED_NUMBER_123_1_OVER_2 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 1, denominator = 2)
+  private val MIXED_NUMBER_123_0_OVER_3 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 0, denominator = 3)
+
+  @Inject
+  internal lateinit var fractionInputHasNoFractionalPartRuleClassifierProvider:
+    FractionInputHasNoFractionalPartRuleClassifierProvider
+
+  private val hasNoFractionalPartClassifierProvider: RuleClassifier by lazy {
+    fractionInputHasNoFractionalPartRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testNoFractionalPart_wholeNumber123Answer_hasNoFractionalPart() {
+    val inputs: Map<String, InteractionObject> = mapOf()
+
+    val matches =
+      hasNoFractionalPartClassifierProvider.matches(
+        answer = WHOLE_NUMBER_123,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNoFractionPart_fraction2Over4Answer_hasFractionalPart() {
+    val inputs: Map<String, InteractionObject> = mapOf()
+
+    val matches =
+      hasNoFractionalPartClassifierProvider.matches(
+        answer = FRACTION_2_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNoFractionPart_mixedNumber123And1Over2Answer_hasFractionPart() {
+    val inputs: Map<String, InteractionObject> = mapOf()
+
+    val matches =
+      hasNoFractionalPartClassifierProvider.matches(
+        answer = MIXED_NUMBER_123_1_OVER_2,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNoFractionPart_mixedNumber123And0Over3Answer_hasNoFractionPart() {
+    val inputs: Map<String, InteractionObject> = mapOf()
+
+    val matches =
+      hasNoFractionalPartClassifierProvider.matches(
+        answer = MIXED_NUMBER_123_0_OVER_3,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNoFractionPart_mixedNumberNegative123And1Over2Answer_hasFractionPart() {
+    val inputs: Map<String, InteractionObject> = mapOf()
+
+    val matches =
+      hasNoFractionalPartClassifierProvider.matches(
+        answer = MIXED_NUMBER_123_1_OVER_2,
```

Comment:
You mentioned `Negative123And1Over2Answer` in the function name, so please create a negative value and use it.